torch/csrc/api/include/torch/nn/modules/sequential.h

```diff
@@ -196,17 +185,23 @@ class SequentialImpl : public Cloneable<SequentialImpl> {
   /// `Sequential(std::make_shared<Module>(3, 4))`.
   template <typename M, typename = torch::detail::enable_if_module_t<M>>
   void push_back(M&& module) {
-    // Need to get rid of any reference components for make_unique.
-    using Type = typename std::remove_reference<M>::type;
-    // Here we move (or copy) the module into a new shared_ptr.
-    push_back(std::make_shared<Type>(std::forward<M>(module)));
```

Comment:
The logic here is moved to https://github.com/pytorch/pytorch/pull/17552/files#diff-3ce0ff9395bf23f007205836f43d31aaR580.